code/modules/mob/living/simple_animal/bot_swarm/swarmer.dm

```diff
@@ -0,0 +1,448 @@
+
+/obj/machinery/portable_atmospherics/canister/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	S << "<span class='warning'>An inhospitable area may be created as a result of destroying this object. Aborting.</span>"
+
+/obj/structure/cable/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	S << "<span class='warning'>Disrupting the power grid would bring no benefit to us. Aborting.</span>"
+
+/obj/machinery/power/apc/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	S << "<span class='warning'>Disrupting the power grid would bring no benefit to us. Aborting.</span>"
+
+/obj/machinery/power/smes/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	S << "<span class='warning'>Disrupting the power grid would bring no benefit to us. Aborting.</span>"
+
+/turf/simulated/wall/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	if(locate(/turf/space) in range(1, src))
+		S << "<span class='warning'>Destroying this object has the potential to cause a hull breach. Aborting.</span>"
+		return
+	..()
+
+/obj/structure/window/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	if(locate(/turf/space) in range(1, src))
+		S << "<span class='warning'>Destroying this object has the potential to cause a hull breach. Aborting.</span>"
+		return
+	..()
+
+/obj/machinery/power/emitter/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)//Stop trying to release the singularity you fuck
+	S << "<span class='warning'>This object is too risky to destroy. Aborting.</span>"
+
+/obj/item/stack/cable_coil/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)//Wiring would be too effective as a resource
+	S << "<span class='warning'>This object does not contain enough materials to work with.</span>"
+
+/obj/machinery/porta_turret/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	S << "<span class='warning'>This machine has been warded against us!</span>"
+
+/mob/living/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	S.DisperseTarget(src)
+
+
+////END CTRL CLICK FOR SWARMERS////
+
+/mob/living/simple_animal/hostile/swarmer/proc/Fabricate(var/atom/fabrication_object,var/fabrication_cost = 0)
+	if(!isturf(loc))
+		src << "<span class='warning'>This is not a suitable location for fabrication. We need more space.</span>"
+	if(resources >= fabrication_cost)
+		resources -= fabrication_cost
+	else
+		src << "<span class='warning'>You do not have the necessary resources to fabricate this object.</span>"
+		return 0
+	new fabrication_object(loc)
+	return 1
+
+/mob/living/simple_animal/hostile/swarmer/proc/Integrate(var/obj/item/target)
+	if(resources >= 100)
+		src << "<span class='warning'>We cannot hold more materials!</span>"
+		return
+	if((target.materials[MAT_METAL]) || (target.materials[MAT_GLASS]))
+		resources++
+		do_attack_animation(target)
+		changeNext_move(CLICK_CD_MELEE)
+		if(istype(target, /obj/item/stack))
+			var/obj/item/stack/S = target
+			S.amount--
+			if(S.amount)
```

Comment:
You should probably use S.use() or some other stack proc, otherwise the stack icon for rods won't be updated for example.